Taskfile.yml

```diff
@@ -0,0 +1,109 @@
+# yaml-language-server: $schema: https://taskfile.dev/schema.json
+
+# Install `task` by following https://taskfile.dev/installation/
+# Run `task up` to start argo.
+
+# - `env PROFILE=mysql task up` to start with MySQL.
+# - `env PROFILE=plugins ARGO_EXECUTOR_PLUGINS=true task up` to start with plugins.
+# - `env PROFILE=sso ARGO_AUTH_MODE=sso task up` to start with SSO.
+
+# The app will be up-and-running between 15s and 1m later (if hot compiled or cold).
+# When using `--watch`, any changes made to the source code will be automatically recompiled and the app restarted, typically within a few seconds.
+
+version: 3
+
+tasks:
+  go-deps:
+    cmd: go mod download
+    # mutex: download
+
+  install:
+    cmd: sh -c "make install PROFILE=$PROFILE"
+    deps: [go-deps]
+    sources: [manifests]
+    env:
+      PROFILE: minimal
+    # mutex: docker
+
+  build-controller:
+    cmd: make ./dist/workflow-controller
+    deps: [go-deps]
+    sources: [cmd/workflow-controller, config, errors, persist, pkg, util, workflow]
+    # mutex: build
+
+  port-forward:
+    cmd: ./hack/port-forward.sh
+    deps: [install]
+    # ports: 9000
```

Comment:
I believe the ports should already be opened, but I might be wrong

Reply:
mm this is not forwarding all the way from my devcontainer to my Mac's localhost, so there may very well be something missing here

Reply:
ok, I misinterpreted the issue above actually. it was indeed forwarding due to the `./hack/port-forward.sh` script, but the UI server actually never started b/c `task` was waiting for the `server` task to complete. The `server` task never completes though as it just runs until it is closed. See [below comment](https://github.com/argoproj/argo-workflows/pull/11928#discussion_r1344883152) for more details